### PR TITLE
docs: update machine-token and status documentation

### PR DESCRIPTION
**Pull request by CalvoM**
## Proposed Commit Message

```
docs: update machine-token and status documentation

We are updating the documentation for the machine-token files, since we added one more world readable
`machine-token.json` file.
```

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [x] I have updated or added any documentation accordingly

### CODE REVIEW

File: dev-docs/references/directory_layout.md [hunk 1]
```diff
@@ -17,4 +17,15 @@ The following describes the intent of UA client related directories:
 | ./apt-conf.d/ | apt config files delivered to /etc/apt/apt-conf.d to automatically allow unattended upgrades of ESM security-related components. If apt proxy settings are configured, an additional apt config file will be placed here to configure the apt proxy. |
 | /etc/ubuntu-advantage/uaclient.conf | Configuration file for the UA client.|
 | /var/lib/ubuntu-advantage/private | `root` read-only directory containing Contract API responses, machine-tokens and service credentials |
+| /var/lib/ubuntu-advantage/machine-token.json | `world` readable directory containing redacted Contract API responses, machine-tokens and service credentials |
```
Comment (lucasmoura) [inline dev-docs/references/directory_layout.md:20]: I think we should change `directory` to `file` here

File: dev-docs/references/directory_layout.md [hunk 2]
```diff
@@ -17,4 +17,15 @@ The following describes the intent of UA client related directories:
...
+
+## Note
+
+We have two `machine-token.json` files, located at:
+- /var/lib/ubuntu-advantage/private/machine-token.json
+- /var/lib/ubuntu-advantage/machine-token.json
+
+The first file, located in the `private` directory, is root read-only. We have another world readable file in the `/var/lib/ubuntu-advantage` directory.
+
+The latter is currently being used when calling the `pro status` command as a non-root user. This file is redacted to remove the sensitive user data.
```
Comment (lucasmoura) [inline dev-docs/references/directory_layout.md:31]: I would change the last part from:

```
to remove the sensitive user data.
```

to

```
to remove any sensitive user data
```

File: dev-docs/references/directory_layout.md (hunk 1 above)
Comment (CalvoM) [reply dev-docs/references/directory_layout.md:20]: Oh yeah, updating.

File: dev-docs/references/what_happens_during_attach.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ After running the command `ua attach TOKEN`, UA will perform the following steps
   token, service credentials, affordances, directives and obligations to allow
   enabling and disabling Ubuntu Advantage services
 * UA client writes the machine token API response to the root-readonly
-  /var/lib/ubuntu-advantage/private/machine-token.json
+  /var/lib/ubuntu-advantage/private/machine-token.json and to the world-readable
+  file /var/lib/ubuntu-advantage/machine-token.json.
```
Comment (orndorffgrant) [inline dev-docs/references/what_happens_during_attach.md:13]:
```suggestion
  /var/lib/ubuntu-advantage/private/machine-token.json and a version with secrets redacted
  to the world-readable file /var/lib/ubuntu-advantage/machine-token.json.
```

**Review by orndorffgrant — APPROVED** [review]
Just a little comment - thanks for the doc updates!

File: dev-docs/references/what_happens_during_attach.md (hunk 3 above)
Comment (CalvoM) [reply dev-docs/references/what_happens_during_attach.md]: Updated.

**Review by lucasmoura — APPROVED** [review]

**Pull request merged**